### DISCUSSION TITLE
4.2.2 Mainboard

### DISCUSSION THREAD

**Discussion by ILordRagnarI** [LeeOtts/Ender3v2-Klipper-Configs#2] [Q&A]
Hey, I wanted to know if it is also possible doing this upgrade with the 4.2.2 Mainboard. I've ordered the Sprite Extruder and already use the BLtouch. Would be nice if I don't have to buy the 4.2.7 Mainboard.

**Comment by LeeOtts** [comment]
They have different pins. So you should be able to take the info from the generic 4.2.2 config file found here, [https://github.com/Klipper3d/klipper/blob/master/config/generic-creality-v4.2.10.cfg](https://github.com/Klipper3d/klipper/blob/master/config/generic-creality-v4.2.10.cfg) and plug it into the corresponding pin on this config. So stepper x for you should be

[stepper_x]
step_pin: PC2
dir_pin: !P89
enable_pin: !PC3
microsteps: 16
rotation_distance: 39.87
endstop_pin: ^PA3
position_endstop: 0
position_max: 235
position_min: -15
homing_speed: 50

Now do that with the rest of the config, and you should be up and running.

**Reply by ILordRagnarI** [reply]
Thank you for your fast reply 😊

**Reply by LeeOtts** [reply]
No problem. Were you able to get it set up?

**Reply by ILordRagnarI** [reply]
Hey, sorry for the late response. First I needed a new pi for this project. Yesterday I finished it. It was easier as expected. Everything is working fine. Did my own pid, input Shaper measurements and so on. Really great project. Respect.

**Reply by metrox-eth** [reply]
Hey LeeOtts, thanks for sharing your work with the community, I'am also using a stock 4.2.2 mobo and tried swapping the pins as you suggested, I used an other config file that worked for me before cuz I was confused by the board version in your example labeled 4.2.10 and I am using 4.2.2. Anyway, it's not working for me, I got this error:

```
mcu 'mcu': Starting serial connect
webhooks client 548018145648: New connection
webhooks client 548018145648: Client info {'program': 'Moonraker', 'version': 'v0.8.0-139-g3d9052d'}
Loaded MCU 'mcu' 105 commands (v0.10.0-548-g2357221b / gcc: (15:7-2018-q2-6) 7.3.1 20180622 (release) [ARM/embedded-7-branch revision 261907] binutils: (2.31.1-11+rpi1+11) 2.31.1)
MCU 'mcu' config: ADC_MAX=4095 BUS_PINS_i2c1=PB6,PB7 BUS_PINS_i2c1a=PB8,PB9 BUS_PINS_i2c2=PB10,PB11 BUS_PINS_spi1=PA6,PA7,PA5 BUS_PINS_spi1a=PB4,PB5,PB3 BUS_PINS_spi2=PB14,PB15,PB13 BUS_PINS_spi3=PB4,PB5,PB3 CLOCK_FREQ=72000000 MCU=stm32f103xe PWM_MAX=255 RECEIVE_WINDOW=192 RESERVE_PINS_serial=PA10,PA9 SERIAL_BAUD=250000 STATS_SUMSQ_BASE=256 STEPPER_BOTH_EDGE=1
mcu 'rpi': Starting connect
mcu 'rpi': Unable to open port: [Errno 2] No such file or directory: '/tmp/klipper_host_mcu'
mcu 'rpi': Unable to open port: [Errno 2] No such file or directory: '/tmp/klipper_host_mcu'
mcu 'rpi': Unable to open port: [Errno 2] No such file or directory: '/tmp/klipper_host_mcu'
```
Here is the modified printer.cfg
[printer.txt](https://github.com/LeeOtts/Ender3v2-Klipper-Configs/files/12640133/printer.txt)

I've tried to restart the pi several times, I wonder what I am doing wrong. It looks like a temporary file is missing. Any help will be appreciated. Thanks in advance.

**Reply by metrox-eth** [reply]
Nevermind I forgot to install the RC script... Looks like I am all set. Thanks for the config.

**Reply by LeeOtts** [reply]
Glad you got it working. Happy Printing!
